### ISSUE TITLE
Inconsistent results between CPU and MPS on MLP.

### ISSUE THREAD

**Issue by Tony-chenjw** [pytorch/pytorch#81384]
### 🐛 Describe the bug

I was running MLP model on MPS, and found the results are inconsistent with the one on CPU (same data, same network parameters and same network architecture). Here is a toy example to show where the error is: (seems like there are someting wrong with the order of indices on MPS tensor calculations.)

```python
import os
import torch
import random
import numpy as np

class EnsembleLinear(torch.nn.Module):
    def __init__(self, in_features, out_features, ensemble_size=3):
        super().__init__()

        self.ensemble_size = ensemble_size

        self.register_parameter('weight', torch.nn.Parameter(torch.ones(ensemble_size, in_features, out_features)))
        self.register_parameter('bias', torch.nn.Parameter(torch.ones(ensemble_size, in_features, out_features)))

    def forward(self, x):
        weight = self.weight
        bias = self.bias

        if len(x.shape) == 2:
            x = torch.einsum('ij,bjk->bik', x, weight)
        else:
            x = torch.einsum('bij,bjk->bik', x, weight)

#         I suppose the error is from x = x + bias, welcome to check it out
#         print(x)
#         print(bias)

        x = x + bias

        return x

device = torch.device("cpu")  # this is for producing results on cpu, in this case, you should comment out next line
# device = torch.device("mps")  # this is for producing results on mps, in this case, you should comment out previous line
model = EnsembleLinear(in_features=3, out_features=3, ensemble_size=3).to(device)
x = torch.arange(3*3, dtype=torch.float32).reshape(3,3).to(device)
y = model(x)
print(y)

# Here as you can see, y on cpu and mps are diffrent:
"""
y on cpu:
tensor([[[ 4.,  4.,  4.],
         [13., 13., 13.],
         [22., 22., 22.]],

        [[ 4.,  4.,  4.],
         [13., 13., 13.],
         [22., 22., 22.]],

        [[ 4.,  4.,  4.],
         [13., 13., 13.],
         [22., 22., 22.]]], grad_fn=<AddBackward0>)

y on mps:
tensor([[[ 4.0000,  4.0000,  4.0000],
         [ 4.0000,  4.0000,  4.0000],
         [ 4.0000,  4.0000,  4.0000]],

        [[13.0000, 13.0000, 13.0000],
         [13.0000, 13.0000, 13.0000],
         [13.0000, 13.0000, 13.0000]],

        [[22.0000, 22.0000, 22.0000],
         [22.0000, 22.0000, 22.0000],
         [22.0000, 22.0000, 22.0000]]], device='mps:0', grad_fn=<AddBackward0>)

"""
```

### Versions

Collecting environment information...
PyTorch version: 1.12.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.4 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 17:00:33)  [Clang 13.0.1 ] (64-bit runtime)
Python platform: macOS-12.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.0
[pip3] torch==1.12.0
[pip3] torchvision==0.13.0
[conda] numpy                     1.23.0                   pypi_0    pypi
[conda] numpy-base                1.22.3           py39h974a1f5_0  
[conda] torch                     1.12.0                   pypi_0    pypi
[conda] torchvision               0.13.0                   pypi_0    pypi

😳 Since it is my first time to post the issue on github, if something is inappropriate, welcome to point it out.
Thanks in advance :)

**Comment by DenisVieriu97** [comment]
Hi @Tony-chenjw - thanks for reporting this issue! This should be fixed in the latest pytorch nightly build (1.13.0.dev20220711). 
Could you please give it a try and let me know if you still see any issue? Thanks!

If you're using pip, you can use `pip3 install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cpu` to install.

**Comment by Tony-chenjw** [comment]
@DenisVieriu97 Thanks a lot! It indeed solves the bug and works fine for me now. 😄
